### PR TITLE
AARA-558: fix submit functionality regression

### PR DESCRIPTION
**Pull request by lauherkcambria**
#### Which User story does this relate to (link)?
-https://osi-cwds.atlassian.net/browse/AARA-558

#### Brief feature description
- fix Regression where SUBMIT button is enabled after successful submission (Implementation of AARA-434 is broken)

#### Does this PR block/effect other stories?
- no

#### Do any commands needs to be run in order to run the project with this PR?
-no

#### Code coverage percent diff as a result of this PR?
-n/a

### CODE REVIEW

**Comment by LeonidMarushevskyi** [comment]
Jenkins retest this please

File: app/javascript/rfa_forms/rfa01a_edit_view/index.js [hunk 1]
```diff
@@ -60,14 +61,12 @@ export default class Rfa01EditView extends React.Component {
...
       const DataValidForSave = !checkForNameValidation(this.state.application.toJS().applicants)
-
-      if (prevState.disableSubmit !== DataValidForSubmit) {
-        this.setState({disableSubmit: DataValidForSubmit})
-        this.setApplicationState('metadata', {'submit_enabled': !DataValidForSubmit})
-      }
-      if (prevState.disableSave !== DataValidForSave) {
+      if (prevState.disableSubmit !== disableSubmit) {
+        this.setState({disableSubmit: disableSubmit})
+      } if (prevState.disableSave !== DataValidForSave) {
```
Comment (nasar-cambria) [inline app/javascript/rfa_forms/rfa01a_edit_view/index.js]: This looks much cleaner. 👍

File: spec/acceptance/rfa_forms/rfa_01a_application_spec.rb [hunk 2]
```diff
@@ -178,11 +178,13 @@
     click_link('Applicant Information')
     click_button('Save Progress')
     click_button 'Submit'
-    # commenting out this test case until Submit button bug is fixed in API
-    # expect(page).to have_button('Submit', disabled: true)
+    expect(page).to have_button('Submit', disabled: true)
+    visit page.driver.current_url
+    expect(page).to have_button('Submit', disabled: true)
+    fill_in('applicants[0].first_name', with: 'G', match: :prefer_exact)
```
[the remark this answers is not in the document]
Comment (lauherkcambria) [reply spec/acceptance/rfa_forms/rfa_01a_application_spec.rb]: to test that it doesn't get enabled after editting once it's already been submitted (as reported as a bug)

File: app/javascript/rfa_forms/rfa01a_edit_view/index.js (hunk 1 above)
Comment (lauherkcambria) [reply app/javascript/rfa_forms/rfa01a_edit_view/index.js]: thanks!

**Comment by lauherkcambria** [comment]
@nasar-cambria I believe i have addressed your comments. nice catch! 👍

**Comment by lauherkcambria** [comment]
jenkins, retest this please.

**Comment by lauherkcambria** [comment]
jenkins, retest this please.

**Pull request merged**